File: Core/src/main/java/dev/lynith/core/ui/theme/dark/ThemeDark.java
package dev.lynith.core.ui.theme.dark;

import dev.lynith.core.ui.components.impl.Button;
import dev.lynith.core.ui.theme.base.ButtonStyles;
import dev.lynith.core.ui.theme.AbstractTheme;

public class ThemeDark extends AbstractTheme {

    public ThemeDark() {
        super("Default Dark", ThemeType.DARK);
    }

    @Override
    public ButtonStyles getButtonStyles(Button button) {
        return new DarkButtonStyles(button);
    }

}

File: Core/src/main/kotlin/dev/lynith/core/ui/theme/dark/ThemeDark.kt
package dev.lynith.core.ui.theme.dark

import dev.lynith.core.ui.components.Component
import dev.lynith.core.ui.components.impl.Button
import dev.lynith.core.ui.styles.ComponentStyles
import dev.lynith.core.ui.theme.AbstractTheme
import dev.lynith.core.ui.theme.base.ButtonStyles

class ThemeDark : AbstractTheme(
    name = "Default Dark",
    type = ThemeType.DARK
) {

    override val colorScheme = DarkColorScheme()

    // Custom styles for components
    override var componentMap: Map<Class<out Component<*, *>>, ComponentStyles<*, *>> = mapOf(
        Button::class.java to ButtonStyles(this),
    )

}
